cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -96,6 +99,56 @@ class CwlWorkflowWomSpec extends FlatSpec with Matchers {
     }
   }
 
+  behavior of "A decoded CWL 3step"
+
+  private val stringOrExpressionTests = Table(
+    ("index", "result"),
+    (0, Coproduct[CommandLineTool.StringOrExpression]("grep")),
+    (1, Coproduct[CommandLineTool.StringOrExpression](refineMV[MatchesECMAScriptExpression]("$(inputs.pattern)"))),
+    (2, Coproduct[CommandLineTool.StringOrExpression](refineMV[MatchesECMAScriptExpression]("$(inputs.file)"))),
+    (3, Coproduct[CommandLineTool.StringOrExpression]("|")),
+    (4, Coproduct[CommandLineTool.StringOrExpression]("wc")),
+    (5, Coproduct[CommandLineTool.StringOrExpression]("-l"))
+  )
+
+  private def getTestName(stringOrExpression: CommandLineTool.StringOrExpression): String = {
+    object StringOrExpressionToTestName extends Poly1 {
+      implicit def caseECMAScriptExpression: Case.Aux[ECMAScriptExpression, String] = {
```

Comment:
Huh, I thought the ascriptions were way worse than this and that was why we weren't including them on `Poly1`s, but this doesn't seem bad to me at all.

Reply:
The default IntellJ ascriptions don't know about Shapeless' [`Aux`-pattern](http://www.vlachjosef.com/aux-pattern-evolution/) and end up explicitly defining the inner `type`. They're all easy enough to type by hand.

Reply:
Very nice, I'd like to continue adding these then.